packages/medusa/src/services/payment-provider.ts

```diff
@@ -274,13 +351,34 @@ export default class PaymentProviderService extends TransactionBaseService {
     sessionInput: Cart | PaymentSessionInput
   ): Promise<PaymentSession> {
     return await this.atomicPhase_(async (transactionManager) => {
-      const provider = this.retrieveProvider(paymentSession.provider_id)
+      const provider = this.retrieveProvider<
+        AbstractPaymentService | AbstractPaymentProcessor
+      >(paymentSession.provider_id)
 
       const context = this.buildPaymentProcessorContext(sessionInput)
 
-      const paymentResponse = await provider
-        .withTransaction(transactionManager)
-        .updatePayment(paymentSession.data, context)
+      let paymentResponse
+      if (provider instanceof AbstractPaymentProcessor) {
+        paymentResponse =
+          (await provider.updatePayment({
+            amount: context.amount,
+            context: context.context,
+            currency_code: context.currency_code,
+            customer: context.customer,
+            email: context.email,
+            billing_address: context.billing_address,
+            resource_id: context.resource_id,
+            paymentSessionData: paymentSession.data,
+          })) ?? {}
```

Comment:
question: Why are we defaulting to an empty object? Could this lead to unexpected side effects e.g. in case we swallow an error from the plugin (and not returning `error`)?

Reply:
if an error is returned, it is an object and therefore the default object only occurs if void is returned